<?xml version="1.0"?>
<launch>
    <!-- Arguments -->
    <arg name="pybullet_gui" default="false"/>
    <arg name="paused" default="true"/>

    <arg name="use_front_iiwa" default="true"/>
    <arg name="use_back_iiwa" default="true"/>
    <arg name="simulation" default="true" />
    <arg name="model_list" default="['iiwas_frame', 'air_hockey_table', 'puck', 'iiwa_front', 'iiwa_back']"/>
    <arg name="table_surface_x" default="1.694"/>
    <arg name="table_surface_y" default="0.813"/>
    <arg name="table_surface_z" default="0.1"/>

    <!-- Description -->
    <include file="$(find air_hockey_description)/launch/air_hockey_description.launch">
    </include>

    <node name="rviz" pkg="rviz" type="rviz"
          args="-d $(find air_hockey_description)/rviz/air_hockey.rviz"/>

    <group ns="iiwa_front">
        <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher" output="screen">
            <param name="publish_frequency" type="double" value="50.0"/>
        </node>
    </group>

    <group ns="iiwa_back">
        <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher" output="screen">
            <param name="publish_frequency" type="double" value="50.0"/>
        </node>
    </group>

    <group ns="iiwas_frame">
        <node pkg="robot_state_publisher" type="robot_state_publisher" name="frame_publisher" output="screen">
            <param name="publish_frequency" type="double" value="50.0"/>
        </node>
    </group>

    <group ns="air_hockey_table">
        <rosparam param="base_position" subst_value="True">[$(arg table_surface_x), $(arg table_surface_y), $(arg table_surface_z)]</rosparam>
        <rosparam param="base_orientation" subst_value="True">[0., 0., 0., 1.]</rosparam>
        <node pkg="robot_state_publisher" type="robot_state_publisher" name="frame_publisher" output="screen">
            <param name="publish_frequency" type="double" value="50.0"/>
        </node>
        <node name="table_tf_publisher" pkg="tf" type="static_transform_publisher"
              args="$(arg table_surface_x) $(arg table_surface_y) $(arg table_surface_z) 0 0 0 1.0 /world /Table 120"/>
    </group>

    <group ns="puck">
        <rosparam param="base_position" subst_value="True">[1.08075, 1.013, $(arg table_surface_z)]</rosparam>
        <node pkg="robot_state_publisher" type="robot_state_publisher" name="frame_publisher" output="screen">
            <param name="publish_frequency" type="double" value="50.0"/>
        </node>
    </group>

    <rosparam file="$(find iiwas_pybullet)/config/iiwa.config" command="load" />
    <rosparam file="$(find air_hockey_pybullet)/config/air_hockey.config" command="load" subst_value="true"/>

    <!-- Simulation -->
    <include file="$(find pybullet_ros)/ros/launch/empty_world.launch">
        <arg name="config_file" value="$(find iiwas_pybullet)/config/environment.config" />
        <arg name="environment" default="iiwas_pybullet.plugins.env_plugin" />
        <arg name="pybullet_gui" default="$(arg pybullet_gui)" /> <!-- if true pybullet gui will show -->
        <arg name="pause_simulation" default="$(arg paused)" /> <!-- if true, will boot with paused physics -->
        <arg name="parallel_plugin_execution" default="False" />
        <arg name="use_deformable_world" default="False" /> <!-- enable/disable soft body simulation -->
        <arg name="gui_options" default="" />
        <arg name="model_list" value="$(arg model_list)"/>
    </include>



</launch>

